codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/protocol/ProtocolParserGenerator.kt

```diff
@@ -0,0 +1,344 @@
+                    customizations,
+                    OperationSection.PopulateErrorMetadataExtras(
+                        customizations,
+                        "generic_builder",
+                        "_response_status",
+                        "_response_headers",
+                    ),
+                )
+                rust("let generic = generic_builder.build();")
+                if (operationShape.operationErrors(model).isNotEmpty()) {
+                    rustTemplate(
+                        """
+                        let error_code = match generic.code() {
+                            Some(code) => code,
+                            None => return Err(#{error_symbol}::unhandled(generic))
+                        };
+
+                        let _error_message = generic.message().map(|msg|msg.to_owned());
+                        """,
+                        "error_symbol" to errorSymbol,
+                    )
+                    withBlock("Err(match error_code {", "})") {
+                        val errors = operationShape.operationErrors(model)
+                        errors.forEach { error ->
+                            val errorShape = model.expectShape(error.id, software.amazon.smithy.model.shapes.StructureShape::class.java)
+                            val variantName = symbolProvider.toSymbol(model.expectShape(error.id)).name
+                            val errorCode = httpBindingResolver.errorCode(errorShape).dq()
+                            withBlock(
+                                "$errorCode => #1T::$variantName({",
+                                "}),",
+                                errorSymbol,
+                            ) {
+                                software.amazon.smithy.rust.codegen.core.rustlang.Attribute.AllowUnusedMut.render(this)
+                                assignment("mut tmp") {
+                                    rustBlock("") {
+                                        renderShapeParser(
+                                            operationShape,
+                                            errorShape,
+                                            httpBindingResolver.errorResponseBindings(errorShape),
+                                            errorSymbol,
+                                            listOf(object : software.amazon.smithy.rust.codegen.core.smithy.customize.OperationCustomization() {
+                                                override fun section(section: software.amazon.smithy.rust.codegen.core.smithy.customize.OperationSection): Writable =
+                                                    software.amazon.smithy.rust.codegen.core.rustlang.writable {
+                                                        if (section is software.amazon.smithy.rust.codegen.core.smithy.customize.OperationSection.MutateOutput) {
+                                                            rust("let output = output.meta(generic);")
```

Comment:
IntelliJ loves these fully qualified imports lol

Reply:
Super annoying... I never notice when it does that too.